Disable User in Microsoft Active Directory with Microsoft Teams (#131)
rapid7/insightconnect-workflows: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md

# Description

> Disable User in Active Directory from Microsoft Teams.

Reviewer: ```suggestion
Disabling a compromised account can limit the scope of an attack and buy valuable time to investigate and contain the threat. This workflow disables a domain user account with a command in Microsoft Teams.
```

Sample Slack Trigger Commands:

`!disable-user <exampleuser>`

# Key Features

> * Disable an Active Directory user directly from Microsoft Teams

Reviewer: ```suggestion
* **Break the Kill Chain** - Credentials provide attackers with easy access to numerous targets. Disabling a compromised account or forcing a password reset can quickly and effectively interrupt an attacker’s kill chain.
* **Buy Time to Investigate and Remediate** - In a pinch, disabling a user account can limit your threat exposure and buy your team valuable time to investigate and contain a threat. 
* **Reduce Portal Fatigue** - You have enough to do without logging into a different solution every 5 minutes. Control your response actions from chat instead.
```

# Requirements

* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
* Active Directory connection

# Documentation

## Setup

Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.

> Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input from the preset text of `change_me`).
> 
> To run the workflow, send the command `!disable-user <exampleuser>` in your configured team and channel. The workflow will post after it has completed.
> In the Team and channel inputs replace `change_me` with the appropriate team and channel.

Reviewer: ```suggestion
Once the workflow has been imported, **each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment!** Edit the input with the preset text of `change_me` in each Teams step in the workflow.

After configuring the Teams steps, activate the workflow in order to trigger it.

# Usage

*This workflow will only trigger in the channel specified in the Microsoft Teams workflow steps.*

To run the workflow, send a message to the specified Microsoft Teams channel starting with the command `!disable-user`.

Commands should be in the following format:

`!disable-user jdoe`

The workflow will post the results in a thread.
```
